src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer_IdRangeAndCategoryValidation.cs

```diff
@@ -118,19 +119,21 @@ static bool ShouldValidateRange((string? prefix, int start, int end) range)
                 // Diagnostic Id '{0}' belonging to category '{1}' is not in the required range and/or format '{2}' specified in the file '{3}'.
                 string arg1 = ruleId;
                 string arg2 = category;
-                string arg3 = string.Empty;
+                var arg3 = new StringBuilder();
                 foreach (var range in allowedIdsInfoList)
                 {
                     if (arg3.Length != 0)
                     {
-                        arg3 += ", ";
+                        arg3.Append(", ");
                     }
 
-                    arg3 += !ShouldValidateRange(range) ? range.prefix + "XXXX" : $"{range.prefix}{range.start}-{range.prefix}{range.end}";
+                    arg3.Append(!ShouldValidateRange(range)
```

Comment:
Should probably use append format or split it into multiple append calls for the interpolated string.

Reply:
You are absolutely right!